GitExtUtils/FileLogging/LogPathGenerator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+
+namespace GitExtUtils.FileLogging
+{
+    public class LogPathGenerator
+    {
+        /// <summary>
+        /// The folder for log files
+        /// </summary>
+        public readonly string LogPath;
+
+        private Tuple<DateTime, string> _fileNameCache = new Tuple<DateTime, string>(DateTime.MinValue, null);
```

Comment:
I'm pretty sure you can make this a named tuple
```suggestion
        private (DateTime Date, string FileName) _fileNameCache = (DateTime.MinValue, null);
```

This makes callsites easier to read:
```cs
        public string GenerateFullPath(DateTime now)
        {
            if (now.Date != _fileNameCache.Date)
            {
                _fileNameCache = Renew(now);
            }

            return _fileNameCache.FileName;
        }
```

Reply:
How are you storing the named tuple for the cache? 

Also note its a private implementation of the class and the class is very small. When things are small it matters much less with naming and abstraction. Like using an `i` in a small `for` loop is totally fine with me. Where with a large implementation naming begins to matter.. a lot.

Reply:
I agree that the ValueTuple is clearer. It's also a a struct so will not be heap allocated.